requirements/pytorch/adjust-versions.py

```diff
@@ -29,34 +28,40 @@ def find_latest(ver: str) -> Dict[str, str]:
     raise ValueError(f"Missing {ver} in {VERSIONS}")
 
 
-def replace(req: str, torch_version: Optional[str] = None) -> str:
+def replace(req: str, torch_version: Optional[str] = None, remove_torch_ver: bool = False) -> str:
     if not torch_version:
         import torch
 
         torch_version = torch.__version__
-    assert torch_version, f"invalid torch: {torch_version}"
+    assert remove_torch_ver or torch_version, f"invalid torch: {torch_version}"
 
     # remove comments and strip whitespace
     req = re.sub(rf"\s*#.*{os.linesep}", os.linesep, req).strip()
 
     latest = find_latest(torch_version)
     for lib, version in latest.items():
-        replace = f"{lib}=={version}" if version else ""
+        if remove_torch_ver and lib == "torch":
+            replace = lib
+        else:
+            replace = f"{lib}=={version}" if version else ""
         req = re.sub(rf"\b{lib}(?!\w).*", replace, req)
 
     return req
 
 
 if __name__ == "__main__":
-    if len(sys.argv) == 3:
-        requirements_path, torch_version = sys.argv[1:]
-    else:
-        requirements_path, torch_version = sys.argv[1], None
-    print(f"requirements_path='{requirements_path}' with torch_version='{torch_version}'")
+    import argparse
```

Comment:
Can you explain why the changes to this file are necessary?

Can't we just skip the step where they are initially fixed, instead of removing them again? I feel like this is a difficult solution in the code to what would be simple in CI

Reply:
This is necessary since a won't be selected if there is any pinned version (even with `--pre`). So I still had to remove any version specifiers.

Reply:
I'll revert this once I have a working docker image